lib/livebook/session/data.ex

```diff
@@ -306,6 +308,13 @@ defmodule Livebook.Session.Data do
     hub_secrets = Livebook.Hubs.get_secrets(hub)
     hub_file_systems = Livebook.Hubs.get_file_systems(hub)
 
+    hub_app_folders =
+      if is_struct(hub, Livebook.Hubs.Team) do
+        Livebook.Teams.get_app_folders(hub)
```

Comment:
Let's make this a contract of Livebook.Hubs, so we avoid `is_struct(hub, Livebook.Hubs.Team)` checks. And we make it so all other hubs return an empty list.